src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/UserTransactionService.java

```diff
@@ -50,21 +61,37 @@ public UserTransaction rulesProcessed(UserTransaction ut) {
     }
 
     @Transactional
-    public UserTransaction commit(UUID id) {
+    public boolean commit(UUID id) {
         UserTransaction ut = userTransactionRepository.findOne(id);
+        if (!canRollbackOrCommit(ut)) {
+            return false;
+        }
         ut.setStatus(UserTransactionStatus.COMMITTED);
-        return userTransactionRepository.save(ut);
+        userTransactionRepository.save(ut);
+        return true;
     }
 
     @Transactional
-    public UserTransaction rollback(UUID id) {
+    public boolean rollback(UUID id) {
         UserTransaction ut = userTransactionRepository.findOne(id);
-
+        if (!canRollbackOrCommit(ut)) {
+            return false;
+        }
         revertChangesManager.revertChanges(ut);
         //todo implement and check optimistic locking using version, don't need to detach entity
         ut.setStatus(UserTransactionStatus.ROLLEDBACK);
+        userTransactionRepository.save(ut);
+        return true;
+    }
 
-        return userTransactionRepository.save(ut);
+    public List<UserTransaction> getTimedOutTransactions() {
+        final OffsetDateTime fiveMinutesAgo = OffsetDateTime.now(clock).minusMinutes(
```

Comment:
Why do we need to base on the 'clock'? Does it have sth to do with Scheduler?

Reply:
it helps with tests and I admit to just follow what I already found in the code